.github/workflows/deploy-onpremise.yml

```
@@ -0,0 +1,99 @@
+          
+          # 헬스체크 (포트는 실제 서비스 포트로 수정)
+          curl -f http://localhost:8080/actuator/health || echo "Health check failed"
+
+  deploy-payment-service:
+    name: Deploy Payment Service
+    runs-on: self-hosted
+    if: contains(github.event.head_commit.modified, 'on-premise/core-payment-service') || github.event_name == 'workflow_dispatch'
+    
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```

actionlint (1.7.10)

[error] the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)